File: RedSheet.DbEntities/Models/RelationshipRequire.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("RelationshipRequires", Schema = "dbo")]
	[Serializable]
	public partial class RelationshipRequire  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int RelationshipRequireId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string RelationshipRequireName { get; set; } 
	[InverseProperty("RelationshipRequire")]
	public ICollection<ProjectBackground> ProjectBackgrounds { get; set; } 


	public RelationshipRequire()
	{
	this.ProjectBackgrounds = new HashSet<ProjectBackground>();

    }

    }
}

File: redsheet-portal/src/app/database-models/relationship-require.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectBackground,  } from './'
export class RelationshipRequire {
    constructor(relationshipRequire?: RelationshipRequire )  {
        let properties = [ "className", "relationshipRequireId", "relationshipRequireName", "projectBackgrounds",];
        for (let property of properties)
            if (relationshipRequire && relationshipRequire[property])
                this[property] = relationshipRequire[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	relationshipRequireId : number =   0 ;
 
    @required()
    @maxLength(200)
	relationshipRequireName : string =   undefined;
	@nested(ProjectBackground)
	projectBackgrounds: ProjectBackground[];



}
